compute_sdk/globus_compute_sdk/sdk/login_manager/authorizer_login_manager.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+import logging
+
+import globus_sdk
+from globus_compute_sdk.sdk.login_manager.manager import LoginManager
+from globus_compute_sdk.sdk.login_manager.protocol import LoginManagerProtocol
+from globus_compute_sdk.sdk.web_client import WebClient
+from globus_sdk.scopes import AuthScopes
+
+from .manager import ComputeScopeBuilder
+
+log = logging.getLogger(__name__)
+
+ComputeScopes = ComputeScopeBuilder()
+
+
+class AuthorizerLoginManager(LoginManagerProtocol):
+    """
+    Implements a LoginManager that can be instantiated with authorizers.
+    This manager can be used to create an Executor with authorizers created
+    from previously acquired tokens, rather than requiring a Native App login
+    flow or Client credentials.
+    """
+
+    def __init__(self, authorizers: dict[str, globus_sdk.RefreshTokenAuthorizer]):
+        self.authorizers = authorizers
+
+    def get_auth_client(self) -> globus_sdk.AuthClient:
+        return globus_sdk.AuthClient(authorizer=self.authorizers[AuthScopes.openid])
+
+    def get_web_client(
+        self, *, base_url: str | None = None, app_name: str | None = None
+    ) -> WebClient:
+        return WebClient(
+            base_url=base_url,
+            app_name=app_name,
+            authorizer=self.authorizers[ComputeScopes.resource_server],
+        )
+
+    def ensure_logged_in(self):
+        """Ensure authorizers for each of the required scopes are present."""
+
+        lm = LoginManager()
+
+        for server, _scopes in lm.login_requirements:
```

Comment:
Looking at the `login_requirements` property, I see:

```python
@property
def login_requirements(self) -> t.Iterator[tuple[str, list[str]]]:
    yield from self.SCOPES.items()
```

And looking at the SCOPES, that's actually a (pragmatically) static class variable:
```python
SCOPES: dict[str, list[str]] = {
    ComputeScopes.resource_server: [ComputeScopes.all],
    AuthScopes.resource_server: [AuthScopes.openid],
}
```

Suggesting that the `login_requirements` property is just a convenience.  However, instantiating the `LoginManager` is nominally a heavyweight action, including:

- creating a GC directory (`_home() / ".globus_compute"`) if it doesn't exist
- connecting to (i.e., sqlite opening) `storage.db`, potentially creating and initializing it

Nominally, not much happens hereafter, but still a steep price for merely wrapping an basically-known-valid token that should not be stored and will be forgotten when the request is done.  Consequently, suggest, for now, to just iterate the SCOPES directly:

```python
for server in LoginManager.SCOPES:
```

Reply:
Good call. I was distracted in trying to align with the existing LM and didn't consider the setup of storage.db and such. I've switched it to just using SCOPES, rather than creating a login manager.